.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,60 @@
+name: Commitlint
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
+  merge_group:
+    branches:
+      - main
+    types:
+      - checks_requested
+jobs:
+  commit-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install conventional-commit linter
+        run: npm install @commitlint/config-conventional @commitlint/cli
+
+      # Default config does not allow sentence case
+      # https://github.com/conventional-changelog/commitlint/blob/master/@commitlint/config-conventional/index.js
+      - name: Add conventional commits config and enable sentence case
+        run: |
+          echo "module.exports = {
+            extends: ['@commitlint/config-conventional'],
+            rules: {
+              'subject-case': [
+                2,
+                'never',
+                ['start-case', 'pascal-case', 'upper-case'],
+              ],
+            }
+          }" > commitlint.config.js
```

Comment:
this makes this config buried in the action, recreating it every time, and also requiring escaping (as it's inside a shell command); please rather save it wouldn't it be better to have it as e.g. `.commitlintrc.js` file in the repository? (according to [1] it seems supported) this way it's easier to review/edit, and it's hidden by default (like `.ansible-lint`, `.yamllint.yml`, etc)

[1] https://github.com/conventional-changelog/commitlint#config

Reply:
My thinking is that since the config is small, it isn't a big deal to just create it in the github action, and save having to create another file in https://github.com/linux-system-roles/.github/tree/main/playbooks/templates and manage it in https://github.com/linux-system-roles/.github/blob/main/inventory/group_vars/active_roles.yml

The reason we have `.ansible-lint` and `.yamllint.yml` are historical - although having an `.ansible-lint` is probably a good thing since it is larger and more complex, but `.yamllint.yml` is almost trivially small.